lighthouse-core/lib/network-recorder.js

```diff
@@ -189,64 +191,86 @@ class NetworkRecorder extends EventEmitter {
    * @param {LH.Crdp.Network.RequestWillBeSentEvent} data
    */
   onRequestWillBeSent(data) {
-    // NOTE: data.timestamp -> time, data.type -> resourceType
-    this.networkManager._dispatcher.requestWillBeSent(data.requestId,
-        data.frameId, data.loaderId, data.documentURL, data.request,
-        data.timestamp, data.wallTime, data.initiator, data.redirectResponse,
-        data.type);
+    const originalRequest = this._findRealRequest(data.requestId);
+    if (originalRequest) {
+      // TODO(phulce): log these to sentry?
+      if (!data.redirectResponse) {
+        return;
+      }
+
+      const modifiedData = {...data, requestId: `${originalRequest.requestId}:redirect`};
+      const redirectRequest = new NetworkRequest();
+
+      redirectRequest.onRequestWillBeSent(modifiedData);
+      originalRequest.onRedirectResponse(data);
+
+      originalRequest.redirectDestination = redirectRequest;
+      redirectRequest.redirectSource = originalRequest;
+
+      this.onRequestStarted(redirectRequest);
+      this.onRequestFinished(originalRequest);
+      return;
+    }
+
+    const request = new NetworkRequest();
```

Comment:
maybe move this case to the early return as it's so much simpler/shorter?

Reply:
done